packages/db/src/index.ts

```diff
@@ -15,12 +15,20 @@ const ACTIONS = {
   SAVE_VAR: "save the {} into the database {} into the table {}",
 };
 
-const parse = (o: string) => {
-  try {
-    return JSON.parse(o);
-  } catch (e) {
-    throw new Error(`[DATABASE] Error in JSON format "${o}"`);
+const parse = (o: any): object => {
+  if (o) {
+    if (typeof o === "object") {
+      return o;
+    }
+    if (typeof o === "string") {
+      try {
+        return JSON.parse(o);
+      } catch (e) {
+        throw new Error(`[DATABASE] Error in JSON format "${o}"`);
+      }
+    }
   }
+  return {};
 };
```

Comment:
I would rather go, with a [EAFP paradigm](https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html) something like:

```
try {
  return JSON.parse(o)
} catch (e) {
  return o ?? {}
}
```
You don't really care about the type of `o`... if it's JSON parsable string then it's returned straight away... if not, if it's not null|undefined with the help of the [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) something is returned ;p
What you think?

Reply:
Omg that is a thing of beaty. Thanks for this tip!